src/Processors/Formats/Impl/ParquetMetadataInputFormat.cpp

```diff
@@ -129,10 +143,35 @@ void checkHeader(const Block & header)
 static std::shared_ptr<parquet::FileMetaData> getFileMetadata(
```

Comment:
There are two implementations of `getFileMetadata`. One for `ParquetBlockInputFormat.cpp` and one for `ParquetMetadataInputFormat`. I am still not sure if I should leave it duplicated or put it somewhere.. The logic is pretty much the same, but requires a couple of arguments to be passed

Reply:
I do not see a lot of duplication (as code is really a bit different). But it depends on whether we are planning to keep this here or move it to upstream later.

If we are keeping it here, I would avoid unnecessary refactoring in this case -- this code does not look like it will change a lot, but changing code in more places may cause additional pain